pkg/sql/logictest/logic.go

```diff
@@ -2966,11 +2973,23 @@ func (t *logicTest) execQuery(query logicQuery) error {
 				// rows in the order in which the query returned them.
 				sortMsg = " -> ignore the following ordering of rows"
 			}
-			fmt.Fprintf(&buf, "but found (query options: %q%s) :\n", query.rawOpts, sortMsg)
-			for _, line := range t.formatValues(actualResultsRaw, query.valsPerLine) {
-				fmt.Fprintf(&buf, "    %s\n", line)
+			var buf bytes.Buffer
+			fmt.Fprintf(&buf, "%s: %s\nexpected:\n%s", query.pos, query.sql, expFormatted.String())
+			fmt.Fprintf(&buf, "but found (query options: %q%s) :\n%s", query.rawOpts, sortMsg, actFormatted.String())
+			if *showDiff {
```

Comment:
maybe we could make it `showDiff || (output greater than 50 lines)` -- or someother similar heuristic?